### DISCUSSION TITLE
New Interface List object

### DISCUSSION THREAD

**Discussion by gautamsi** [keystonejs/keystone#5333] [Feature requests]
I wish if we can create List by extending a List class. This is what I want to do.

1. Create list like a Class rather than just bunch of field config. 
2. Add my own methods to this class so that I can have business logic together with lists rather than scattered all over. 
3. Able to retrieve the instance of the class by using `keystone.lists.MyList`
4. invoke async methods on the list to complete the run logic, for example if I have relationship items in 1:many relation, I can retrieve the list instance and invoke business logic against the item id. Business logic would decide if this can work with id or need more field and how to get that with context or adapter

I am also thinking if the list instance can be per item (tied to id)

random thoughts, I was working on custom graphql and thought it would be much easier if I had custom method on the list itself.

**Comment by bladey** [comment]
Thanks for sharing this @gautamsi.
